### PR TITLE
Feature/api overview

### PR DESCRIPTION
**Pull request by Aurora12**

### CODE REVIEW

File: store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -3021,3 +3029,64 @@ func (s SqlChannelStore) GetWorkChannels(teamId string, userId string) (*model.C
 func (s SqlChannelStore) GetGlobalChannels(teamId string, userId string) (*model.ChannelSnapshotList, *model.AppError) {
 	return s.getSpecificChannels(teamId, userId, "Kind = '' AND Type = 'O'")
 }
+
+func (s SqlChannelStore) GetOverview(teamId string, userId string) (*model.ChannelList, *map[string][]string, *[]string, *model.AppError) {
```
Comment (ygworldr) [inline store/sqlstore/channel_store.go:3033]: Could you add a comment? You know, public API and all that…

File: store/storetest/channel_store.go [hunk 2]
```diff
@@ -4440,3 +4441,14 @@ func testChannelStoreGetGlobalChannels(t *testing.T, ss store.Store) {
 		assert.True(t, v.Channel.TeamId == teamId)
 	}
 }
+
```
Comment (ygworldr) [inline store/storetest/channel_store.go:4444]: Do we need a failure test?

File: store/sqlstore/channel_store.go (hunk 1 above)
Comment (Aurora12) [reply store/sqlstore/channel_store.go:3033]: ⚡

File: store/storetest/channel_store.go (hunk 2 above)
Comment (Aurora12) [reply store/storetest/channel_store.go:4444]: I don't think it is applicable here. If there is no content for the request it just returns empty lists, that's all. What kind of failure can we test for there?

**Pull request merged**